eth/handler_eth.go

```diff
@@ -181,6 +190,11 @@ func (h *ethHandler) handleBlockAnnounces(peer *eth.Peer, hashes []common.Hash,
 // handleBlockBroadcast is invoked from a peer's message handler when it transmits a
 // block broadcast for the local node to process.
 func (h *ethHandler) handleBlockBroadcast(peer *eth.Peer, block *types.Block) error {
+	// Do not handle any broadcast until we finish resetting from the bad state.
+	// This should be a very small time window
+	if h.Core().BadHashExistsInChain() {
+		return nil
```

Comment:
And here, we expect this to be rare but would be helpful if something does go wrong.

Reply:
added warning